src/Entities/ArmorStand.cpp

```diff
@@ -0,0 +1,461 @@
+	m_CustomNameAlwaysVisible(false),
+	m_Size(a_Size),
+	m_HasArms(false),
+	m_HasBasePlate(true),
+	m_HeadRotation(0.0, 0.0, 0.0),
+	m_BodyRotation(0.0, 0.0, 0.0),
+	m_LeftArmRotation(-10.0, 0.0, -10.0),
+	m_RightArmRotation(-15.0, 0.0, 10.0),
+	m_LeftLegRotation(-1.0, 0.0, -1.0),
+	m_RightLegRotation(1.0, 0.0, 1.0),
+	m_LeftHand(cItem()),
+	m_RightHand(cItem()),
+	m_Helmet(cItem()),
+	m_ChestPlate(cItem()),
+	m_Leggings(cItem()),
+	m_Boots(cItem())
+{
+	SetYaw(a_Yaw);
+	SetMaxHealth(1);
+	SetHealth(1);
+}
+
+
+
+
+
+void cArmorStand::SpawnOn(cClientHandle & a_ClientHandle)  // Should got any rotation and should replace any of non-solid blocks like snow
+{
+	a_ClientHandle.SendSpawnObject(*this, 78, 0, static_cast<Byte>(GetYaw()), static_cast<Byte>(GetPitch()));
+	a_ClientHandle.SendEntityMetadata(*this);
+	a_ClientHandle.SendEntityEquipment(*this, 0, GetEquippedWeapon());
+	a_ClientHandle.SendEntityEquipment(*this, 1, GetEquippedBoots());
+	a_ClientHandle.SendEntityEquipment(*this, 2, GetEquippedLeggings());
+	a_ClientHandle.SendEntityEquipment(*this, 3, GetEquippedChestplate());
+	a_ClientHandle.SendEntityEquipment(*this, 4, GetEquippedHelmet());
+	a_ClientHandle.SendEntityEquipment(*this, 5, GetOffHandEquipedItem());  // For compatibility issues, left hand was added at the end and not at the ID 1 like wants the protocol since 1.9
+}
+
+
+
+
+
+void cArmorStand::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
+{
+	UNUSED(a_Dt);
+	UNUSED(a_Chunk);
+	// I don't know how and why but by default the client considere it as no gravity, so disable gravity on the server unless it fixed totally and configurable
+	// Gravity for such entity is not really important
+}
+
+
+
+
+
+void cArmorStand::KilledBy(TakeDamageInfo & a_TDI)
+{
+	// TODO: Replace harm by shaking effect when hurt
+
+	if (IsMarker())  // Disallow interaction with marker
+	{
+		return;
+	}
+
+	cItems Pickup;
+	if (!m_LeftHand.IsEmpty())
+	{
+		Pickup.push_back(m_LeftHand);
+		SetOffHandEquipedItem(cItem());
+	}
+	else if (!m_RightHand.IsEmpty())
+	{
+		Pickup.push_back(m_RightHand);
+		SetEquippedWeapon(cItem());
+	}
+	else if (!m_Helmet.IsEmpty())
+	{
+		Pickup.push_back(m_Helmet);
+		SetEquippedHelmet(cItem());
+	}
+	else if (!m_ChestPlate.IsEmpty())
+	{
+		Pickup.push_back(m_ChestPlate);
+		SetEquippedChestplate(cItem());
+	}
+	else if (!m_Leggings.IsEmpty())
+	{
+		Pickup.push_back(m_Leggings);
+		SetEquippedLeggings(cItem());
+	}
+	else if (!m_Boots.IsEmpty())
+	{
+		Pickup.push_back(m_Boots);
+		SetEquippedBoots(cItem());
+	}
+	else
+	{
+		super::KilledBy(a_TDI);
+		Pickup.push_back(cItem(E_ITEM_ARMOR_STAND));
+		Destroy();
+	}
+	GetWorld()->SpawnItemPickups(Pickup, GetPosX(), GetPosY() + 1, GetPosZ());
+	if (IsTicking())
+	{
+		SetHealth(GetMaxHealth());
+		SetInvulnerableTicks(0);
+		GetWorld()->BroadcastEntityMetadata(*this);
+	}
+}
+
+
+
+
+
+void cArmorStand::OnRightClicked(cPlayer & a_Player)  // Only works for tags, else the callback is not called, IDK Why ...
```

Comment:
I have figured out why your right click handler isn't being called. The armour stand has its own variant of the use entity packet which isn't handled by the protocol yet.
https://github.com/cuberite/cuberite/blob/8b4d37771f6948f57f9e2d6359c574c2b411cc05/src/Protocol/Protocol_1_9.cpp#L2787-L2793

Reply:
Thanks a lot ! I had solved it locally, you will see it nextly